feat: add jsonbinpack preset (#854)
asyncapi/modelina: docs/languages/TypeScript.md

Check out this [example out for a live demonstration](../../examples/typescript-generate-marshalling).

## Generate models with jsonbinpack support

This functionality is for the library [jsonbinpack](https://github.com/sourcemeta/jsonbinpack).

This preset can ONLY be used with AsyncAPI 2.x and JSON Schema draft 4 to 7 inputs.

This functionality has two requirements:
1. You MUST manually install the library `jsonbinpack`.
2. You MUST also use the [Generate un/marshal functions for classes](#generate-unmarshal-functions-for-classes)

> This feature enables you to conver the models into a buffer that is highly space efficient, this is instead of sending pure JSON data across the wire.

Reviewer: ```suggestion
This feature allows you to convert models to a buffer, which is highly space-efficient, instead of sending pure JSON data over the wire.
```

> Check out this [example out for a live demonstration](../../examples/typescript-generate-jsonbinpack/).

Reviewer: Hmm, live demonstration sentence is usually used for browser app, so maybe

```suggestion
Check out this [example](../../examples/typescript-generate-jsonbinpack/), which uses a preset.
```

Author: This is what we use across all other docs, I am all up for changing it to something else, but let's stay consistent 🙂
